### ISSUE TITLE
Issues with proxied @import https urls 

### ISSUE THREAD

**Issue by wzrdtales** [jakubpawlowicz/clean-css#741]
# Problem description

If you're using a http proxy the current implementation causes trouble and renders it completely unusable.

Example error:

```
File: assets/vendor.css
Broken @import declaration of "https://fonts.googleapis.com/css?family=Lato:400,700,400italic,700italic&subset=latin" - write EPROTO
```

Gist: https://gist.github.com/wzrdtales/621ff62d38bb94737215

# Details

The failure is here;

https://github.com/jakubpawlowicz/clean-css/blob/master/lib/imports/inliner.js#L271-L273

This can't work and wont work if there is only a http proxy, but not a https one. And this is common for mostly all the time, to have just a http proxy.

# Possible solutions

Two options to fix this:

- Fix the check for http and https in combination with the proxy.
- Use the `request` lib instead

Using the `request` lib would be also be more convenient for the users, as it automatically parses `http_proxy` and `https_proxy` environment variables and handles many edge cases already. Thus the users never need to set any proxy configurations for `clean-css` if they are already defined them in their environment.

**Comment by jakubpawlowicz** [comment]
Could you please provide a test case with CSS and system proxy settings? I'll try to replicate it locally first.

**Comment by wzrdtales** [comment]
@jakubpawlowicz Really thought the error message is already enough to reproduce it.

Here is a gist for you https://gist.github.com/wzrdtales/621ff62d38bb94737215

**Comment by jakubpawlowicz** [comment]
Thank you! It's always easier to reproduce an issue with a code snippet to make sure we are on the same page. Can see it failing now.

**Comment by jakubpawlowicz** [comment]
Released in 3.4.11, thanks @wzrdtales!
